pkg/graph/resolvers/model_plan.go

```diff
@@ -33,83 +30,68 @@ func ModelPlanCreate(logger *zap.Logger, modelName string, store *storage.Store,
 	}
 
 	// Create an initial collaborator for the plan
-	collab := &models.PlanCollaborator{
-		ModelPlanRelation: models.ModelPlanRelation{
-			ModelPlanID: createdPlan.ID,
-		},
-		EUAUserID: principalInfo.EuaUserID,
-		FullName:  principalInfo.CommonName,
-		TeamRole:  models.TeamRoleModelLead,
-		Email:     principalInfo.Email.String(),
-		BaseStruct: models.BaseStruct{
-			CreatedBy: principalInfo.EuaUserID,
-		},
-	}
+	collab := models.NewPlanCollaborator(principal.ID(), createdPlan.ID)
+	collab.EUAUserID = principalInfo.EuaUserID
+	collab.FullName = principalInfo.CommonName
+	collab.TeamRole = models.TeamRoleModelLead
+	collab.Email = principalInfo.Email.String()
```

Comment:
I like this, I think collaborator is a prime example of an instance we could / should do this. I'll take a look in other places and see if it can be implemented elsewhere as well.

Reply:
I found Discussion and Discussion Reply for other structs that should have other parameters for the constructor. I updated those, ModelPlan and PlanCollaborator to take more parameters. I think PlanDocuments might benefit from this treatment as well, but I'm excluding it for now. 

It is treated very differently than elsewhere in the code, so I think tackling this when we look at refactoring document upload would be more approriate.

Reply:
Yeah, [this PR](https://github.com/CMSgov/mint-app/pull/220) also _heavily_ modifies Plan Documents, so I wouldn't worry about it for that reason too.